src/systems/media-pdf-oculus-fix.js

```diff
@@ -0,0 +1,53 @@
+/*
+  PDFjs which is used to render pdfs depends on the window.requestAnimationFrame function.
+  This is not called when in mobileVR.
+  This system replaces the function with the requestAnimationFrame of the xrSession.
+  It also adds a listener to ensure it is reset and call not called callbacks after the user leaves xr.
+*/
+export class MediaPDFOculusFix {
+  constructor() {
+    this.requestAnimationFramePrev = this.requestAnimationFramePrev || window.requestAnimationFrame;
+    this.cbList = new Set();
+  }
+
+  tick(sceneEl, exit) {
+    const isMobileVR = AFRAME.utils.device.isMobileVR();
+    let isVR = !exit && sceneEl.is("vr-mode") && isMobileVR;
+    if (isVR != this.wasVR) {
+      if (isVR) {
+        this.onEnterVR(sceneEl);
+      } else if (this.wasVR) {
+        this.onExitVR();
+      }
+    }
+    this.wasVR = isVR;
+  }
+
+  onEnterVR(sceneEl) {
+    const { xrSession } = sceneEl;
+    window.requestAnimationFrame = cb => {
+      let myCb = () => {
+        this.cbList.delete(myCb);
+        cb();
+      };
+      this.cbList.add(myCb);
+      return xrSession.requestAnimationFrame.call(xrSession, myCb);
+    };
+    //Make sure tick is called
+    if (!this.interval)
+      this.interval = setInterval(() => {
+        this.tick(sceneEl);
+      }, 500);
```

Comment:
Not sure why this `setInterval` is needed, `this.mediaPDFOculusFix.tick(this.el)` should already be called right?

Reply:
That `setInterval` was maybe a previous fix before you patch `window.requestAnimationFrame`?
And that 500ms was to match the default minDurationToUpdateCanvas?
https://github.com/mozilla/pdf.js/blob/264046736d6cf3b726a111eb1461b01e56e774b2/web/base_pdf_page_view.js#L62C11-L62C36